src/snowflake/connector/cursor.py

```diff
@@ -1273,6 +1311,18 @@ def wait_until_ready():
                     )
                 )
             self._inner_cursor.execute(f"select * from table(result_scan('{sfqid}'))")
+
+            if (
+                self._inner_cursor._total_rowcount == 1
+                and self._inner_cursor.fetchall()
```

Comment:
`fetchall` does affect its cursor's results, so if `[("Multiple statements executed successfully.",)]` isn't what `fetchall` returns then you've affected what the cursor will return